radicle-httpd/src/lib.rs

```diff
@@ -95,8 +88,60 @@ pub async fn run(options: Options) -> anyhow::Result<()> {
         )
         .into_make_service_with_connect_info::<SocketAddr>();
 
-    axum::Server::bind(&options.listen)
+    axum::Server::bind(&listen)
         .serve(app)
         .await
         .map_err(anyhow::Error::from)
 }
+
+/// Create a router consisting of other sub-routers.
+fn router(options: Options, home: Home) -> anyhow::Result<Router> {
+    let profile = Arc::new(radicle::Profile::load_at(home)?);
+
+    tracing::info!("using radicle home at {}", profile.home().display());
+
+    let ctx = api::Context::new(profile.clone());
+
+    let api_router = api::router(ctx);
+    let git_router = git::router(profile.clone(), options.aliases);
+    let raw_router = raw::router(profile);
+
+    let app = Router::new()
+        .merge(git_router)
+        .nest("/api", api_router)
+        .nest("/raw", raw_router);
+
+    Ok(app)
+}
+
+#[cfg(test)]
+mod routes {
+    use std::collections::HashMap;
+    use std::net::SocketAddr;
+
+    use axum::extract::connect_info::MockConnectInfo;
+    use axum::http::StatusCode;
+
+    use radicle::profile::Home;
+
+    use crate::test::{self, get};
+
+    #[tokio::test]
+    async fn test_invalid_route_returns_404() {
+        let tmp = tempfile::tempdir().unwrap();
+        let _ctx = test::seed(tmp.path());
+        let app = super::router(
+            super::Options {
+                aliases: HashMap::new(),
+                listen: SocketAddr::from(([0, 0, 0, 0], 8080)),
+            },
+            Home::new(tmp.path().join("radicle")).unwrap(),
```

Comment:
sub-routes take a `ctx` which has profile in it, and are designed to be easily testable, and here, to be able to re-produce the same system we have to pass `home` ourselves. 

for this to work, I had to also add `load_at(home: Home) -> Result<Self, Error>` to [profile.rs](https://github.com/radicle-dev/heartwood/blob/03c636c65d757ab7008ee3c65963dd585979dc64/radicle/src/profile.rs).